adapters/taboola/taboola.go

```diff
@@ -89,6 +106,42 @@ func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.R
 	return bidResponse, errs
 }
 
+func (a *adapter) buildRequest(request *openrtb2.BidRequest) (*adapters.RequestData, error) {
+	requestJSON, err := json.Marshal(request)
+	if err != nil {
+		return nil, err
+	}
+
+	// MediaType set based on first imp
+	mediaType := "display"
+	if request.Imp[0].Native != nil {
+		mediaType = "native"
+	}
```

Comment:
Optional improvement: when you execute `buildRequest` function you already know what type is it. Instead of L116-L119 just pass  "display" or "native" from `MakeRequests`. Also please extract "display" and "native" to constants

Reply:
@VeronikaSolovei9  You are totally right! 
just committed the change :)